### ISSUE TITLE
SM mapped area missing/corrupted

### ISSUE THREAD

**Issue by Josephur** [tewtal/alttp_sm_combo_randomizer#23]
Console or Emulator: Snes9x 1.55

Original Filename: Not available (renamed file, sorry)
Seed: Not available

Item, Entrance or Custom:  Used fortune teller near links house to return to SM after finding Gravity suit in ALttP

Description of problem:  First I realized I've already been to this area before, why is the map data missing, then upon closer inspection some of it was there, but then if you look at the downward hill in the map, only parts of it in the look mapped (purple not blue), which is impossible.

See attached screenshot.

![image](https://user-images.githubusercontent.com/6666442/40820327-df1a3e3c-652d-11e8-84e2-a5cc0038dc61.png)

**Comment by lenophis** [comment]
It's possible the routine to recall the map is pulling from the wrong area. It'll be looked into.

**Comment by Josephur** [comment]
Have experienced this 3- times over different seeds now (9.2), just wanted to update you that it happens pretty frequently.  Not a huge deal for tournament style players, but for us casuals knowing where we've been is a huge help.

**Comment by MP2E** [comment]
Can confirm that this happens to me on v9.2 as well, over a few seeds. I'm using an sd2snes and 1chip SNES

**Comment by lenophis** [comment]
Yeah, it has been confirmed. I know why it's happening, just haven't been able to nail down a fix yet. Coming soon!

**Comment by alercah** [comment]
Is the problem that some map data is getting ported from one area to another when doing a game warp?

**Comment by lenophis** [comment]
We have found out why. It *should* be fixed for v10.2. We're just testing it to make sure it works properly, and doesn't break anything else.
